pkg/builtins/ec_op.go

```
@@ -0,0 +1,262 @@
+	}
+
+	//All input cells should be filled, and be integer values
+	//If an input cell is not filled, return None
+
+	input_cells := make([]lambdaworks.Felt, 0)
+	for i := 0; i < int(ec.n_input_cells); i++ {
+		addr, err := segments.Get(instance.AddUint(uint(i)))
+		if err == nil {
+			felt, is_felt := addr.GetFelt()
+			if is_felt {
+				input_cells = append(input_cells, felt)
+			} else {
+				return nil, errors.New("Runner error, Expected Integer for input cells")
+			}
+		} else {
+			return nil, nil
+		}
+	}
+
+	for j := 0; j < 2; j++ {
+		x := input_cells[EC_POINT_INDICES[j].x]
+		y := input_cells[EC_POINT_INDICES[j].y]
+		if !PointOnCurve(x, y, alpha, beta) {
+			return nil, errors.New("Point not in curve")
+		}
+	}
+
+	prime, ok := new(big.Int).SetString(PRIME[2:], 16)
+	if !ok {
+		return nil, errors.New("Could not parse prime")
+	}
+
+	alpha_big_int := big.NewInt(1)
+
+	partial_sum := PartialSum{X: input_cells[0], Y: input_cells[1]}
+	double_point := DoublePoint{X: input_cells[2], Y: input_cells[3]}
+
+	result, err := EcOnImpl(partial_sum, double_point, input_cells[4], alpha_big_int, prime, ec.scalar_height)
+
+	felt_result_x := lambdaworks.FeltFromHex(result.X.Text(16))
+	felt_result_y := lambdaworks.FeltFromHex(result.Y.Text(16))
+
+	ec.cache[x_addr] = felt_result_x
+	ec.cache[x_addr.AddUint(1)] = felt_result_y
+
+	if index-uint(ec.n_input_cells) == 0 {
+		return memory.NewMaybeRelocatableFelt(felt_result_x), nil
+	} else {
+		return memory.NewMaybeRelocatableFelt(felt_result_y), nil
+	}
+}
+
+func LineSlope(point_a PartialSumB, point_b DoublePointB, prime big.Int) (big.Int, error) {
+	mod_value := new(big.Int).Sub(&point_a.X, &point_b.X)
+	mod_value.Mod(mod_value, &prime)
+
+	if mod_value.Cmp(big.NewInt(0)) == 0 {
+		return big.Int{}, errors.New("is multiple of prime")
+	}
+
+	n := new(big.Int).Sub(&point_a.Y, &point_b.Y)
+	m := new(big.Int).Sub(&point_a.X, &point_b.X)
+
+	z, _ := math_utils.DivMod(n, m, &prime)
```

Dont ignore the error